layer/filestore.go

```diff
@@ -97,16 +97,20 @@ func (fm *fileMetadataTransaction) SetCacheID(cacheID string) error {
 	return ioutil.WriteFile(filepath.Join(fm.root, "cache-id"), []byte(cacheID), 0644)
 }
 
-func (fm *fileMetadataTransaction) TarSplitWriter() (io.WriteCloser, error) {
+func (fm *fileMetadataTransaction) TarSplitWriter(compress bool) (io.WriteCloser, error) {
```

Comment:
At first I thought this argument was controlling whether the file got compressed data or uncompressed data. The meaning would be clearer if the name was changed to alreadyCompressed and the logic was inverted.